### ISSUE TITLE
how i can see the progress of a train execution in jupyter notebook?

### ISSUE THREAD

**Issue by amperesz** [ultralytics/yolov5#12223]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have to train a model, and when I run train.py in Google Colab, I can see the training progress. However, when I run the jupyter notebook in an Anaconda environment, the progress is not visible. My code has been running for over an hour and hasn't printed anything. What could be wrong?

![image](https://github.com/ultralytics/yolov5/assets/68832627/89179815-58e8-4ef8-8028-1b14680acb33)

i expect something like this:

![image](https://github.com/ultralytics/yolov5/assets/68832627/46cbcdf0-7c99-441f-af5e-dc11ebf63533)

### Additional

I have already installed the requirements.txt file.

the command i run is:

`!python yolov5/train.py --img 640 --batch 16 --epochs 100 --data yolov5/data.yaml --cfg yolov5/models/yolov5s.yaml --weights yolov5/yolov5s.pt --cache`

**Comment by glenn-jocher** [comment]
@amperesz in order to see the progress of a train execution in a Jupyter Notebook, you can enable the logging feature in YOLOv5. By default, the logging level is set to "info", but you can increase the verbosity to get more detailed progress updates.

To do so, modify your code as follows:

```python
import logging
logging.basicConfig(level=logging.DEBUG)  # Add this line before running yolov5/train.py
```

By adding the `logging.basicConfig(level=logging.DEBUG)` line of code before running `yolov5/train.py`, you will enable the debug level logging, which will print more detailed progress updates during the training process.

Please give this a try and let us know if you have any further questions.

**Comment by amperesz** [comment]
hi @glenn-jocher 
it has been running for 15 minutes and there is no difference

![image](https://github.com/ultralytics/yolov5/assets/68832627/786fa683-cc8c-4433-93d0-d2d6a7d64d2e)

**Comment by glenn-jocher** [comment]
Hi there @amperesz! Thank you for reaching out. 

It looks like the training process has been running for 15 minutes without any noticeable progress. There could be a few reasons for this. 

First, please check the size of your dataset. If you have a large dataset, it may take some time to process and train the model. Also, make sure that your training parameters such as batch size, number of epochs, and learning rate are set appropriately for your dataset and model.

Additionally, please ensure that your hardware, such as the CPU or GPU, is properly configured and has sufficient resources to handle the training workload. Sometimes, insufficient resources can cause delays in the training process.

If you've reviewed these aspects and are still experiencing issues, please provide more details about your dataset, training parameters, and hardware configuration so that we can better assist you.

**Comment by glenn-jocher** [comment]
Hi @amperesz, thanks for reaching out.

I'm here to help! It looks like the training process has been running for 15 minutes without any noticeable progress. There could be a few reasons for this. 

First, please verify that the training process is actually running and hasn't encountered any errors that are preventing progress updates from being displayed.

Second, ensure that your training parameters, such as batch size, number of epochs, and learning rate, are configured appropriately for your specific dataset and model.

If everything appears to be in order, please share more details about your setup, such as your dataset size, hardware configuration, and any error messages or logs that may provide additional insight into the issue.

I'm here to help troubleshoot and get your training process back on track!

**Comment by glenn-jocher** [comment]
👋 Hello @amperesz, thank you for reaching out.

It seems that the training process has been running for 15 minutes without any noticeable progress. The lack of progress might be due to various reasons such as incorrect configuration, hardware limitations, or issues with the dataset.

To troubleshoot this, please ensure that your training configurations are suitable for your dataset and hardware. Additionally, consider checking if there are any errors being logged in the console that could be causing the delay in progress.

If you can provide more details about your dataset, training parameters, and hardware setup, it will help in identifying the root cause of the issue and providing a more specific solution.

Let me know if you need further assistance or have any other questions!
